### PR TITLE
Dockerfile: updated bases to 4.0 

### PR DESCRIPTION
**Pull request by bryanlatten**
- Github Actions: replacing Docker hub with multi-arch build

Requirements (pre-merge)
- [x] Add docker hub credentials
- [x] Disable Docker hub autobuild.

For the moment, Travis CI will continue PR validation testing (but is currently broken due to Docker rate limiting)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+          sleep 5
+          curl localhost:${VANILLA_PORT} | grep "${{ env.TEST_MATCH }}"
+          curl localhost:${ENV_FILE_PORT} | grep "${{ env.TEST_MATCH }}"
+          curl -k https://localhost:${HTTPS_PORT} | grep "${{ env.TEST_MATCH }}"
+
+          docker kill $CONTAINER_VANILLA
+          docker kill $CONTAINER_ENV_FILE
+          docker kill $CONTAINER_HTTPS
+          docker kill CONTAINER_REGISTRY
```
Comment (bossjones) [inline .github/workflows/ci.yml:67]: new line

**Comment by adobejmong** [comment]
I use https://github.com/behance/docker-base/tree/master/.github/workflows to help me compare what we have working on docker-base. Eye-balling it looks correct to me. The main difference between base and this is the split between `ci` and `publish`. 

It might be worthwhile to add those comments in the README itself under the `Release Management` section. For example, you clearly had to separate it due to the port and cert section in the `ci` (which isn't necessarily present in base).

Other than that it seems OK to me. 👍

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          if [ -n "$v" ]; then
+            echo "Injected docker socket detected"
+            TARGET_HOST="host.docker.internal"
+          elif [ -S /var/run/docker.sock ]; then
+            TARGET_HOST="localhost"
+          else
+            echo "No Docker socket detected, fail"
+            exit 1
+          fi
+          echo TARGET_HOST=$TARGET_HOST >> $GITHUB_ENV
```
Comment (bossjones) [inline .github/workflows/ci.yml:41]: echo "TARGET_HOST=${TARGET_HOST}" >> "${GITHUB_ENV}"

**Comment by adobejmong** [comment]
👍

File: container/root/scripts/fix_woff_support.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+# Detects if woff support is already present
+if grep -Fxq "font/woff" /etc/nginx/mime.types
+then
+  echo "Woff type detected, no changes necessary"
+else
+  echo "Woff type not detected, adding..."
+  sed -i "s/}/\n    font\/woff                             woff;&/" /etc/nginx/mime.types
+  sed -i "s/}/\n    font\/woff2                            woff2;\n&/g" /etc/nginx/mime.types
+fi
```
Comment (bossjones) [inline container/root/scripts/fix_woff_support.sh:14]: only missing a newline then we are good to go !

**Comment by bryanlatten** [comment]
Unclear why PR status isn't recording correctly. Here is the passing Travis tests: https://github.com/behance/docker-nginx/pull/78/checks?check_run_id=4526170286.

Merging!

**Pull request merged**